connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java

```diff
@@ -116,6 +108,66 @@ public void testDeltaPathTable() throws IOException, ApiException {
         session.stop();
     }
 
+    @Test
+    public void testManagedParquetTable() throws ApiException, IOException, ParseException {
+        createCommonResources();
+        SparkSession session = createSparkSessionWithCatalogs(SPARK_CATALOG);
+        // Spark only allow `spark_catalog` to return built-in file source tables.
+        setupManagedParquetTable(PARQUET_TABLE, new ArrayList<>(0), session);
+        String tableFullName = SPARK_CATALOG + "." + SCHEMA_NAME + "." + PARQUET_TABLE;
+
+        Dataset<Row> df = session.sql("SELECT * FROM " + tableFullName);
+        FileSourceScanExec scan = (FileSourceScanExec) df.queryExecution().executedPlan().collectLeaves().head();
+        Assert.assertEquals("test1", scan.relation().options().apply("fs.s3a.access.key"));
+        Assert.assertEquals("test2", scan.relation().options().apply("fs.s3a.secret.key"));
+        Assert.assertEquals("test3", scan.relation().options().apply("fs.s3a.session.token"));
```

Comment:
This is not really testing whether all file system access are going to use these tokens. 
I suggest a different approach completely. 
- Make UC vend different credentials (fake ones) for different paths. 
- Create a fake FileSystem object that is used for S3 paths, but actually it's a wrapper for LocalFileSystem. With that you can intercept all of file system operations that Spark will do (including file listing, etc.)
- For each of those operations, throw error if operations on path prefix X is not configured with credentials for path X. 

If the spark read operation passes, then we will have confirmed that all file system operations have been correctly configured to use the right s3 credentials. 

Does that make sense?

Reply:
We probably need to update the UC sever to make it more testable. Right now the managed table location must starts with `s3://` so I can't really use a fake FileSystem to test it. Can we allow managed table location to be local file system for the ease of testing?